src/controllers/weather_controller.py

```diff
@@ -40,4 +40,43 @@ def get_weather_for_growth_period(self, barley_type, planting_date, lat, lon):
             harvest_date = planting_date + timedelta(days=270)  # Approx. 9 months
         return self.get_historical_weather(planting_date.strftime("%Y-%m-%d"), harvest_date.strftime("%Y-%m-%d"), lat, lon)
 
+def extract_daily_data(self, data):
+    result = []
+
+    dates = data[0]["date"]
+    max_temps = data[0]["max_temperature"]
+    min_temps = data[0]["min_temperature"]
+    precipitations = data[0]["precipitation"]
+
+    current_day_data = {}
+
+    for i in range(len(dates)):
```

Comment:
I'm just wondering if there is an API endpoint to directly get data for that day instead of looping through data.